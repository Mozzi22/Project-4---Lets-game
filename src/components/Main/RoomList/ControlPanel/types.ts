export type TModal = {
    isOpen: boolean; 
    data: {}, 
    modalType: string;
}
export interface IControlPanel {
    setValue: () => void; 
    filterByRoomName: () => void; 
    changeModalVisibility: (arg: TModal) => void;
}
