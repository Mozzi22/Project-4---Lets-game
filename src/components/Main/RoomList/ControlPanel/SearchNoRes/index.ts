import SearchNoRes from './SearchNoRes';

export default SearchNoRes;
