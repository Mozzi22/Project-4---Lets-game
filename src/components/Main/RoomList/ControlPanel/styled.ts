import styled from 'styled-components';

export const StControlPanel = styled.div`
    display: flex;
    border-top: 1px solid black;
    border-bottom: 1px solid black;    
`;
export const StCreateRoom = styled.div`

`;
