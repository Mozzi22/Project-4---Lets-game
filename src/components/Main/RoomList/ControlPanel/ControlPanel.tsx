import React from 'react';
import Button from 'src/components/UI/Button';
import Input from 'src/components/UI/Input';
import { useTranslation } from 'react-i18next';
import { StControlPanel } from './styled';
import { IControlPanel } from './types';

const ControlPanel: React.FC<IControlPanel> = ({ setValue, filterByRoomName, changeModalVisibility }) => {
    const { t } = useTranslation();
    const handleCreateRoom = () => changeModalVisibility({
        isOpen: true, data: {}, modalType: 'createGame',
    }); 
    return ( 
        <StControlPanel>
            <Input
                id='nameOfYourId'
                height="60px"
                key='nameOfYourId'
                name="filterByRoomName"
                inputHeight="60px"
                borderRadius="0px"
                value={filterByRoomName}
                onChange={setValue}
                placeholder='placeholder_search_input'
                margin="0 auto" 
                color="white"
                bgColor="transparent"
                fontSizeInp="20px"
                borderColor="transparent"
                padding="17px"
            />
                <Button
                    id='create_room'
                    name='create_room'
                    onClick={handleCreateRoom}
                    content={t('create_room')}
                    width="100px"
                    height="60px"
                />
        </StControlPanel> 
    );
};

export default ControlPanel;
