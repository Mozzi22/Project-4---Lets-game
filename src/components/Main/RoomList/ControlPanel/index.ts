import { connect } from 'react-redux';
import ControlPanel from './ControlPanel';
// import { rooms } from '/src/Store/chat/selectors';
import { changeModalVisibility } from 'src/Store/modals/actions';

// const mapStateToProps = state => ({ rooms: rooms(state) });
const mapDispatchToProps = dispatch => ({
    changeModalVisibility: payload => dispatch(changeModalVisibility(payload)),
});

export default connect(null, mapDispatchToProps)(ControlPanel);
