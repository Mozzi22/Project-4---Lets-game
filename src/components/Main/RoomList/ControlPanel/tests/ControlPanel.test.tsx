import React from 'react';
import ControlPanel from '../ControlPanel';
import { mount, shallow } from 'enzyme';

describe('ControlPanel', () => {
    const props = {
        setValue: jest.fn(),
        filterByRoomName: jest.fn(),
        changeModalVisibility: jest.fn()
    }
    it('Should match snapshot', () => {
        const component = shallow(<ControlPanel {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StControlPanel', () => {
        const component = mount(<ControlPanel {...props}/>);
        expect(component.find('styled__StControlPanel')).toHaveLength(1);
    });
     it('should render Input', () => {
        const component = mount(<ControlPanel {...props}/>);
        expect(component.find('Input')).toHaveLength(1);
    });
    it('should render Button', () => {
        const component = mount(<ControlPanel {...props}/>);
        expect(component.find('Button')).toHaveLength(1);
    });
    it('should call handleCreateRoom after click Button', () => {
        const component = mount(<ControlPanel {...props}/>);
        component.find('button').simulate('click');
        expect(props.changeModalVisibility).toHaveBeenCalledWith({
            isOpen: true, data: {}, modalType: 'createChat',
        });
    });
});
