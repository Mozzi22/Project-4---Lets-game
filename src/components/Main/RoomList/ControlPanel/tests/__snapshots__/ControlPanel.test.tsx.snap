// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanel Should match snapshot 1`] = `"<div class=\\"styled__StControlPanel-sc-17c3c8y-0 ftvLLF\\"><div height=\\"60px\\" class=\\"styled__StInputContainer-sc-1hxx3n2-3 jGunPj\\"><input type=\\"text\\" id=\\"nameOfYourId\\" name=\\"filterByRoomName\\" color=\\"white\\" placeholder=\\"placeholder_search_input\\" class=\\"styled__StInput-sc-1hxx3n2-1 fLRxDY\\"/></div><button id=\\"create_room\\" name=\\"create_room\\" type=\\"button\\" height=\\"60px\\" width=\\"100px\\" class=\\"styled__StButton-sc-15atqtm-0 dNhglT\\">create_room</button></div>"`;
